staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2191,17 +2191,41 @@ type SMBiosConfiguration struct {
 
 // MigrationConfiguration holds migration options
 type MigrationConfiguration struct {
-	NodeDrainTaintKey                 *string            `json:"nodeDrainTaintKey,omitempty"`
-	ParallelOutboundMigrationsPerNode *uint32            `json:"parallelOutboundMigrationsPerNode,omitempty"`
-	ParallelMigrationsPerCluster      *uint32            `json:"parallelMigrationsPerCluster,omitempty"`
-	AllowAutoConverge                 *bool              `json:"allowAutoConverge,omitempty"`
-	BandwidthPerMigration             *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
-	CompletionTimeoutPerGiB           *int64             `json:"completionTimeoutPerGiB,omitempty"`
-	ProgressTimeout                   *int64             `json:"progressTimeout,omitempty"`
-	UnsafeMigrationOverride           *bool              `json:"unsafeMigrationOverride,omitempty"`
-	AllowPostCopy                     *bool              `json:"allowPostCopy,omitempty"`
-	DisableTLS                        *bool              `json:"disableTLS,omitempty"`
-	Network                           *string            `json:"network,omitempty"`
+	// NodeDrainTaintKey defines the taint key that indicates a node should be drained.
+	// Note: this option relies deprecated node taint feature. Default: kubevirt.io/drain
+	NodeDrainTaintKey *string `json:"nodeDrainTaintKey,omitempty"`
+	// ParallelOutboundMigrationsPerNode is the maximum number of concurrent outgoing live migrations
+	// allowed per node. Defaults to 2
+	ParallelOutboundMigrationsPerNode *uint32 `json:"parallelOutboundMigrationsPerNode,omitempty"`
+	// ParallelMigrationsPerCluster it the total number of concurrent live migrations
+	// allowed cluster-wide. Defaults to 5
+	ParallelMigrationsPerCluster *uint32 `json:"parallelMigrationsPerCluster,omitempty"`
+	// AllowAutoConverge allows the platform to compromise performance/availability of VMIs to
+	// guarantee successful VMI live migrations. Defaults to false
+	AllowAutoConverge *bool `json:"allowAutoConverge,omitempty"`
+	// BandwidthPerMigration limits the amount of network bandwith live migrations are allowed to use.
+	// The value is in MiB per second. Defaults to 0 (no limit)
+	BandwidthPerMigration *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
+	// CompletionTimeoutPerGiB is the maximum number of seconds 1GiB can take to live-migrate.
+	// If, during any CompletionTimeoutPerGiB period, less than 1GiB is transfered from the source
+	// to the target, the migration is cancelled. Defaults to 800
```

Comment:
I think we should also mention that it determines the amount of time takes for post-copy to kick in if enabled:
https://github.com/kubevirt/kubevirt/blob/1f7c616daac748e6b11660920514eb8613d71ec2/pkg/virt-launcher/virtwrap/live-migration-source.go#L623-L625

Reply:
Interesting, I did not know that!
PTAL, too see if I got it right.

Reply:
Look great!
Maybe:
```
unless AllowPostCopy is true, which will enforce the migration to change its mode to post-copy
```

just for some more clarifications?